### ISSUE TITLE
RISCV purecap uses cfromptr instead of csetaddr on NULL

### ISSUE THREAD

**Issue by arichardson** [CTSRD-CHERI/llvm-project#383]
```
 i8 addrspace(200)* inttoptr (i64 32 to i8 addrspace(200)*)
```
results in
```
addi a0, zero, 32
cfromptr cs0, ddc, a0
```

MIPS generates `cncoffset	$c18, $cnull, 32`. I would expect RISCV to also use CSetAddr/CIncOffsetImm here.

**Comment by jrtc27** [comment]
Yes, this is a known bug. Fixing it for purecap is trivial, the question is what to do in hybrid, but I think we decided that it should also do a cincoffset on null, and require the frontend to emit the intrinsic it wants for the semantics it expects?
